pkg/connection/connection.go

```diff
@@ -43,12 +43,19 @@ func New(opts Options, creds Credentials) *ApiConnection {
 }
 
 func (conn ApiConnection) BuildRequest(verb string, path string, body any) (*http.Request, error) {
-	bodyData, err := json.Marshal(body)
+	buffer := bytes.Buffer{}
+
+	// Make sure we do not encode HTML data which might interfere with instance_data which can be valid
+	// XML.
+	encoder := json.NewEncoder(&buffer)
+	encoder.SetEscapeHTML(false)
```

Comment:
It turns out that escaping is enabled by default. This interferes with the `xml/json` document style sent by public cloud clients. So I disabled it, any thoughts how this could backfire?

Reply:
Not that I can think of right now...